docs/internals/constants.md

```diff
@@ -0,0 +1,147 @@
+(constants-note)=
+
+# Handling of closed-over constants
+
+"Closed-over constants" are non-scalar arrays that are encountered during JAX tracing
+of a function and do not have dependencies on any of the function's arguments.
+JAX operations such as `jax.numpy` and `lax` are staged out and do not create
+closed-over constants.
+In the following example, the arrays
+`a_jax_array` and `np.full` are closed-over constants, but `jnp.full`
+is not. We refer below to closed-over constants simply as constants.
+
+```python
+from jax import numpy as jnp
```

Comment:
nit:

```suggestion
import numpy as np
from jax import jit
from jax import numpy as jnp
```

which would make the code runnable if copy-pasted

since `jax` and `np` are used later in the example:

```
@jit
 def f(x):
   return x + a_jax_array + np.full((16,), 42.) + jnp.full((16,), 142.)
```